lib/fsmx.ex

```diff
@@ -2,15 +2,17 @@ defmodule Fsmx do
   @moduledoc """
   """
 
-  @spec transition(struct(), binary()) :: {:ok, struct} | {:error, any}
+  @type state :: binary() | atom()
```

Comment:
The convention I know is about [using t()](https://elixirforum.com/t/what-exactly-is-t-except-type-specification/33786/3). 
Since it is inside a @spec, it should be clear that we are in types territory. 
An example using a similar signature: https://github.com/sasa1977/parent/blob/master/lib/parent/client.ex#L25